charon/src/bin/charon-driver/translate/translate_constants.rs

```diff
@@ -144,21 +144,13 @@ impl<'tcx, 'ctx> BodyTransCtx<'tcx, 'ctx> {
                 RawConstantExpr::Ref(Box::new(val))
             }
             ConstantExprKind::Cast { .. } => {
-                raise_error!(
-                    self,
-                    span,
-                    "Unsupported constant: `ConstantExprKind::Cast {{..}}`",
-                )
+                RawConstantExpr::Opaque("`ConstantExprKind::Cast {{..}}`".into())
```

Comment:
I'm ok with returning this, but we should still emit a warning imo. At least current Charon policy is to emit warnings when we can't trnaslate something. FOr that, replace the `raise_error` above with `register_error`.